src/ngraph_backend_manager.h

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <tuple>
+
+#include "ngraph/runtime/backend.hpp"
+
+namespace tensorflow {
+namespace ngraph_bridge {
+
+// Manages the nGraph bridge's notion of the current execution backend.
+class BackendManager {
+ public:
+  // Returns the global BackendManager instance.
+  //
+  // (Note that in testing scenarios, there may be additional instances.)
+  static BackendManager* Instance();
```

Comment:
This is a factory method - so need to delete all the other constructors or mark private with test class declared as friend if needed for testing purposes.